src/Crawler.php

```diff
@@ -323,7 +332,15 @@ public function setUserAgent(string $userAgent): Crawler
 
     public function getUserAgent(): string
     {
-        return $this->client->getConfig('headers')['User-Agent'];
+        $headers = $this->client->getConfig('headers');
+
+        foreach (array_keys($headers) as $name) {
+            if (strtolower($name) === 'user-agent') {
+                return (string) $headers[$name];
+            }
+        }
+
+        return static::DEFAULT_USER_AGENT;
```

Comment:
If we didn't need `DEFAULT_USER_AGENT` previously, couldn't we simply throw an exception here? No user agent in the client's config seems like something we don't need a graceful fallback for.

Reply:
Why should we throw an exception if the user agent header is not required? This would break the `LinkAdder::shouldCrawl()` method. Better solution is to change return type declaration to `?string` and return `null` when there is no user agent. Do you agree?

Reply:
You're right, let's return `*`

Reply:
Instead of removing constant I've only changed it's value to avoid magic string.